src/rulesets/Base/array.jl

```diff
@@ -103,3 +103,52 @@ function rrule(::typeof(fill), value::Any, dims::Int...)
     end
     return fill(value, dims), fill_pullback
 end
+
+#####
+##### `repeat`
+#####
+
+function rrule(::typeof(repeat), x::AbstractVector, m::Integer)
+    function repeat_pullback(Ȳ)
+        return (NO_FIELDS, dropdims(sum(reshape(Ȳ, length(x), :); dims=2); dims=2), DoesNotExist())
+    end
+    return repeat(x, m), repeat_pullback
+end
+
+function rrule(::typeof(repeat), x::AbstractVecOrMat, m::Integer, n::Integer=1)
+    function repeat_pullback(Ȳ)
+        Ȳ′ = reshape(Ȳ, size(x, 1), m, size(x, 2), n)
+        return (NO_FIELDS, reshape(sum(Ȳ′; dims=(2,4)), size(x)), DoesNotExist(), DoesNotExist())
+     end
+    return repeat(x, m, n), repeat_pullback
+ end
+
+function rrule(::typeof(repeat), xs::AbstractArray; inner=ntuple(_->1, ndims(xs)), outer=ntuple(_->1, ndims(xs)))
+    function repeat_pullback(Ȳ)
+        Ȳ′ = zero(xs)
+        S = size(xs)
+        for (dest_idx, val) ∈ pairs(IndexCartesian(), Ȳ)
+            src_idx = [mod1(div(dest_idx[dim] - 1, inner[dim]) + 1, S[dim]) for dim ∈ 1:length(S)]
+            Ȳ′[src_idx...] += val
+        end
+        return (NO_FIELDS, Ȳ′)
+    end
+    return repeat(xs; inner=inner, outer=outer), repeat_pullback
+end
+
+function rrule(::typeof(repeat), x::AbstractArray{<:Real, 0}, m::Integer)
+    repeat_pullback(Ȳ) = (NO_FIELDS, similar(x, eltype(Ȳ)) .= sum(Ȳ), DoesNotExist())
+    return repeat(x, m), repeat_pullback
+end
+
+function frule((_, Δx), ::typeof(repeat), x, m::Integer)
+    return repeat(x, m), repeat(Δx, m)
+end
+
+function frule((_, Δxs), ::typeof(repeat), xs; inner=ntuple(_->1, ndims(xs)), outer=ntuple(_->1, ndims(xs)))
+    return repeat(xs; inner=inner, outer=outer), repeat(Δxs; inner=inner, outer=outer)
+end
+
+function frule((_, Δx), ::typeof(repeat), x::AbstractArray{<:Real,0}, m::Integer)
+    return repeat(x, m), repeat(fill(Δx,m))
+end
```

Comment:
These are the rules not covered by #460, which only did `rrules`. PR could be rebased to still add these.

Reply:
I will take a look at this around Wednesday this week after finishing my late JuliaCon vid. Sorry I lost track of them

Reply:
No rush, was just closing stale PRs & thought this one should get a note for whenever someone looks.